### PR TITLE
add words to lesson

### PR DESCRIPTION
**Pull request by phuongbk**

### CODE REVIEW

File: app/views/lessons/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<%= form_for @lesson, url: category_lesson_path(@lesson.category, @lesson), method: "patch", html: {id: "question_form"} do |f| %>
+  <h1><%= @lesson.category.name %></h1>
+  <h2>
+    <span id="answer_count">1</span>/
+    <span id="total"><%= @lesson.lesson_words.count %></span>
+  </h2>
+  <%= f.fields_for :lesson_words do |ff| %>
+    <div class="row answer-form">
+      <div class="col-md-6">
+        <p class="alert alert-info"><%= ff.object.word.content   %></p>
```
Comment (phonghenry14) [inline app/views/lessons/edit.html.erb]: thừa dấu cách ở cuối

File: app/models/lesson.rb [hunk 2]
```diff
@@ -4,4 +4,16 @@ class Lesson < ActiveRecord::Base
 
   has_many :lesson_words, dependent: :destroy
   has_many :words, through: :lesson_words, dependent: :destroy
+
+  after_save :add_word
```
Comment (phonghenry14) [inline app/models/lesson.rb]: call back để sau các quan hệ

File: app/models/answer.rb [hunk 3]
```diff
@@ -5,4 +5,6 @@ class Answer < ActiveRecord::Base
   has_many :lesson, through: :lesson_words
 
   validates :content, presence: true
+
+  accepts_nested_attributes_for :lesson_words
```
Comment (phonghenry14) [inline app/models/answer.rb]: validate cũng để sau quan hệ

File: app/models/lesson.rb [hunk 4]
```diff
@@ -4,4 +4,16 @@ class Lesson < ActiveRecord::Base
 
   has_many :lesson_words, dependent: :destroy
   has_many :words, through: :lesson_words, dependent: :destroy
+
+  accepts_nested_attributes_for :lesson_words
+
+  after_save :add_word
```
Comment (dieunb) [inline app/models/lesson.rb]: try `before_save`

File: config/routes.rb [hunk 5]
```diff
@@ -21,4 +21,11 @@
   resources :categories, only: :index
   resources :lessons, only: :create
   match "/users/:id/:type", to: "relationships#index", via: :get
+
+  resources :categories, only: :index do
+    resources :lessons do
+      resources :lesson_words
```
Comment (dieunb) [inline config/routes.rb]: don't user nested resources three levels

File: app/models/word.rb [hunk 6]
```diff
@@ -6,6 +6,18 @@ class Word < ActiveRecord::Base
   has_many :lessons, through: :lesson_words
 
   accepts_nested_attributes_for :answers
+  accepts_nested_attributes_for :lesson_words
```
Comment (dieunb) [inline app/models/word.rb]: use `reject_if` to handle some exception

File: app/models/answer.rb [hunk 7]
```diff
@@ -4,5 +4,8 @@ class Answer < ActiveRecord::Base
   has_many :lesson_words
   has_many :lesson, through: :lesson_words
 
+  accepts_nested_attributes_for :lesson_words
```
Comment (dieunb) [inline app/models/answer.rb]: no need

File: app/models/lesson.rb [hunk 8]
```diff
@@ -4,4 +4,16 @@ class Lesson < ActiveRecord::Base
 
   has_many :lesson_words, dependent: :destroy
   has_many :words, through: :lesson_words, dependent: :destroy
+
+  accepts_nested_attributes_for :lesson_words
```
Comment (dieunb) [inline app/models/lesson.rb]: use `reject_if` to handle any exception

File: app/models/word.rb [hunk 9]
```diff
@@ -6,6 +6,18 @@ class Word < ActiveRecord::Base
   has_many :lessons, through: :lesson_words
 
   accepts_nested_attributes_for :answers
```
Comment (dieunb) [inline app/models/word.rb]: use `reject_if` and `allow_destroy` to handle any exception

File: app/models/category.rb [hunk 10]
```diff
@@ -2,6 +2,8 @@ class Category < ActiveRecord::Base
   has_many :lessons, dependent: :destroy
   has_many :words, dependent: :destroy
 
+  accepts_nested_attributes_for :lessons
```
Comment (dieunb) [inline app/models/category.rb]: no need

File: app/models/lesson.rb [hunk 11]
```diff
@@ -4,4 +4,16 @@ class Lesson < ActiveRecord::Base
 
   has_many :lesson_words, dependent: :destroy
   has_many :words, through: :lesson_words, dependent: :destroy
+
+  accepts_nested_attributes_for :lesson_words, reject_if: lambda { |a| a["content"].blank? }
```
Comment (dieunb) [inline app/models/lesson.rb]: convention

File: app/models/lesson.rb [hunk 12]
```diff
@@ -4,4 +4,16 @@ class Lesson < ActiveRecord::Base
...
+
+  before_save :add_word
+
+  private
+  def add_word
+    category.words.random(user).each do |word|
```
Comment (dieunb) [inline app/models/lesson.rb]: try `words = category.words.random(user)`

File: app/models/word.rb [hunk 13]
```diff
@@ -5,7 +5,19 @@ class Word < ActiveRecord::Base
   has_many :lesson_words
   has_many :lessons, through: :lesson_words
 
-  accepts_nested_attributes_for :answers
+  accepts_nested_attributes_for :answers, allow_destroy: true,
+   reject_if: lambda { |a| a["content"].blank? }
```
Comment (dieunb) [inline app/models/word.rb]: convention

File: app/models/lesson.rb [hunk 14]
```diff
@@ -4,4 +4,14 @@ class Lesson < ActiveRecord::Base
...
   has_many :lesson_words, dependent: :destroy
   has_many :words, through: :lesson_words, dependent: :destroy
+
+  accepts_nested_attributes_for :lesson_words, reject_if: lambda {|a| a["content"].blank?}
+
+  before_save :add_word
+
+  private
+  def add_word
+    self.words = category.words.random(user)
```
Comment (dieunb) [inline app/models/lesson.rb]: try remove `self` and recheck

File: app/models/word.rb [hunk 15]
```diff
@@ -5,7 +5,19 @@ class Word < ActiveRecord::Base
   has_many :lesson_words
   has_many :lessons, through: :lesson_words
 
-  accepts_nested_attributes_for :answers
+  accepts_nested_attributes_for :answers, allow_destroy: true,
+   reject_if: lambda {|a| a["content"].blank?}
```
Comment (dieunb) [inline app/models/word.rb]: `attributes` not `a`

**Pull request merged**